### PR TITLE
Add AWS For Fluent Bit module

### LINKED ISSUE #104: Add FluentBit log forwarding to CloudWatch
In order to have the ability to forward logs to CloudWatch, we need to have additional module for it

### PR DESCRIPTION
**Pull request by DementevNikita**
Fixes #104.

Adds module for deploying AWS For Fluent Bit.

Possible output plugins are:
* CloudWatch
* Kinesis Data Stream
* Kinesis Firehose
* ElasticSearch cluster

### CODE REVIEW

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Create namespace logging
+resource "kubernetes_namespace" "logging" {
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:2]: Script down if namespace exist (use get namespace.name logging and if exist = count 0)

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+resource "helm_release" "aws-for-fluent-bit" {
+  depends_on = [
+    var.module_depends_on
+  ]
+
+  name       = "aws-for-fluent-bit"
+  repository = "https://aws.github.io/eks-charts"
+  chart      = "aws-for-fluent-bit"
+  version    = "0.1.3"
+  namespace  = var.namespace_name
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:20]: you create namespace but set namespace by variable?

File: modules/logging/aws-for-fluent-bit/variables.tf [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+  default     = ""
+  description = "Log record key value of which will be sent to CloudWatch. By default whole message will be sent"
+}
+
+variable "cloudwatch_log_format" {
+  default     = ""
+  description = "Format of the log data"
+}
+
+variable "cloudwatch_role_arn" {
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/variables.tf:54]: Could you please create role and policy for all output options and auto setup this arn here?

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 1 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf]: Done

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 2 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf]: Fixed

File: modules/logging/aws-for-fluent-bit/variables.tf (hunk 3 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/variables.tf]: Done

**Comment by DementevNikita** [comment]
@akastav, Cloudwatch Logs, Firehose and KDS plugins have been tested.

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+resource "aws_iam_role" "aws_for_fluent_bit_iam_role" {
+  name               = "${var.cluster_name}-aws-for-fluent-bit"
+  assume_role_policy = data.aws_iam_policy_document.aws_for_fluent_bit_assume_role_policy.json
+
+  tags = {
+    Environment = var.environment
+    Project     = var.project
+  }
+}
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:38]: Could you please move this sections to iam.tf?

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+      cloudwatch_match                         = var.cloudwatch_match
+      cloudwatch_log_group_name                = var.cloudwatch_log_group_name
+      cloudwatch_log_stream_name               = var.cloudwatch_log_stream_name
+      cloudwatch_log_stream_prefix             = var.cloudwatch_log_stream_prefix
+      cloudwatch_log_key                       = var.cloudwatch_log_key
+      cloudwatch_log_format                    = var.cloudwatch_log_format
+      cloudwatch_auto_create_group             = var.cloudwatch_auto_create_group
+      cloudwatch_endpoint                      = var.cloudwatch_endpoint
+      cloudwatch_credentials_endpoint          = var.cloudwatch_credentials_endpoint
+      cloudwatch_cross_account_role_arn        = var.cloudwatch_cross_account_role_arn
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:76]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+      firehose_enabled                         = var.firehose_enabled
+      firehose_match                           = var.firehose_match
+      firehose_delivery_stream                 = var.firehose_delivery_stream
+      firehose_data_keys                       = var.firehose_data_keys
+      firehose_endpoint                        = var.firehose_endpoint
+      firehose_time_key                        = var.firehose_time_key
+      firehose_time_key_format                 = var.firehose_time_key_format
+      firehose_cross_account_role_arn          = var.firehose_cross_account_role_arn
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:85]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/main.tf [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      kinesis_data_keys                        = var.kinesis_data_keys
+      kinesis_endpoint                         = var.kinesis_endpoint
+      kinesis_sts_endpoint                     = var.kinesis_sts_endpoint
+      kinesis_time_key                         = var.kinesis_time_key
+      kinesis_time_key_format                  = var.kinesis_time_key_format
+      kinesis_compression                      = var.kinesis_compression
+      kinesis_aggregation                      = var.kinesis_aggregation
+      kinesis_experimental_concurrency         = var.kinesis_experimental_concurrency
+      kinesis_experimental_concurrency_retries = var.kinesis_experimental_concurrency_retries
+      kinesis_cross_account_role_arn           = var.kinesis_cross_account_role_arn
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/main.tf:101]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+cloudWatch:
+  enabled: ${cloudwatch_enabled}
+  region: ${aws_region}
+  match: ${cloudwatch_match}
+  logGroupName: ${cloudwatch_log_group_name}
+  logStreamName: ${cloudwatch_log_stream_name}
+  logStreamPrefix: ${cloudwatch_log_stream_prefix}
+  logKey: ${cloudwatch_log_key}
+  logFormat: ${cloudwatch_log_format}
+  roleArn: ${cloudwatch_cross_account_role_arn}
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:61]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+  endpoint: ${cloudwatch_endpoint}
+  credentialsEndpoint: ${cloudwatch_credentials_endpoint}
+
+firehose:
+  enabled: ${firehose_enabled}
+  region: ${aws_region}
+  match: ${firehose_match}
+  deliveryStream: ${firehose_delivery_stream}
+  dataKeys: ${firehose_data_keys}
+  roleArn: ${firehose_cross_account_role_arn}
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:72]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+kinesis:
+  enabled: ${kinesis_enabled}
+  region: ${aws_region}
+  match: ${kinesis_match}
+  stream: ${kinesis_stream}
+  partitionKey: ${kinesis_partition_key}
+  appendNewline: ${kinesis_append_new_line}
+  dataKeys: ${kinesis_data_keys}
+  roleArn: ${kinesis_cross_account_role_arn}
```
Comment (akastav) [inline modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:85]: This params and variables still needs?

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 4 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf]: Done

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 5 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf:48]: In a case of single account - no, cloudwatch_cross_account_role_arn variable is not needed, but in a cross-account case it is, that's why I left it here.

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 6 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf:57]: https://github.com/provectus/swiss-army-kube/pull/105#discussion_r523717989

File: modules/logging/aws-for-fluent-bit/main.tf (hunk 7 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/main.tf:73]: https://github.com/provectus/swiss-army-kube/pull/105#discussion_r523717989

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml (hunk 8 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:61]: https://github.com/provectus/swiss-army-kube/pull/105#discussion_r523717989

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml (hunk 9 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:72]: https://github.com/provectus/swiss-army-kube/pull/105#discussion_r523717989

File: modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml (hunk 10 above)
Comment (DementevNikita) [reply modules/logging/aws-for-fluent-bit/values/aws-for-fluent-bit.yaml:85]: https://github.com/provectus/swiss-army-kube/pull/105#discussion_r523717989

**Review by akastav — APPROVED** [review]

**Pull request merged**